### DISCUSSION TITLE
How to ignore inline links to "Headings" in same document using "remark-renumber-references"

### DISCUSSION THREAD

**Discussion by AnweshGangula** [Xunnamius/unified-utils#15] [Q&A]
I would like to use [`remark-renumber-references`](https://www.npmjs.com/package/remark-renumber-references) in a CLI to format the ReadMe files of my GitHub repositories. But when I use it, the package considers all links - even the references to the same page like `[Example](#Heading3)`.

I would like the package to ignore links like these and only number external links - links that are outside of the current page.

How to achieve this?

Please note that below command is what I'm using:

```
remark -o --use renumber-references README.md 
```

**Comment by Xunnamius** [comment]
Hmm. It wouldn't be much work to add a config flag for that, though have you considered just going with all reference links? There's something to be said for the consistency of such an approach.

If you're trying to prevent only a specific set of links from being transformed (like auto-generated stuff and/or a table of contents), you can do that with [remark-ignore](https://npm.im/remark-ignore). I use it to preserve inline links for doctoc sections and headings with links in them ([example](https://raw.githubusercontent.com/Xunnamius/unified-utils/main/packages/remark-renumber-references/README.md)).
